packages/verified-fetch/test/abort-handling.spec.ts

```diff
@@ -159,39 +125,137 @@ describe('abort-handling', function () {
     expect(blockRetriever.retrieve.callCount).to.equal(1)
   })
 
-  it('should abort a request during unixfs.stat call', async function () {
+  it('should abort a request while loading a file root', async function () {
     const fs = unixfs(helia)
-    const fileCid = await fs.addBytes(Uint8Array.from([0, 1, 2, 3]))
+
+    // add a file with a very small chunk size - this is to ensure we end up
+    // with a DAG that contains a root and some leaf nodes
+    const fileCid = await fs.addBytes(Uint8Array.from([0, 1, 2, 3]), {
+      chunker: fixedSize({ chunkSize: 2 })
+    })
     const directoryCid = await fs.addDirectory()
     const cid = await fs.cp(fileCid, directoryCid, 'index.html')
 
-    await expect(makeAbortedRequest(verifiedFetch, [cid], unixFsStatCalled.promise)).to.eventually.be.rejectedWith('aborted')
+    const leaf1 = CID.parse('bafkreifucp2h2e7of7tmqrns5ykbv6a55bmn6twfjgsyw6lqxolgiw6i2i')
+    const leaf2 = CID.parse('bafkreihosbapmxbudbk6a4h7iohlb2u5lobrwkrme4h3p32zfv2qichdwm')
 
-    expect(peerIdResolver.callCount).to.equal(0) // not called because parseResource never passes the resource to parseUrlString
-    expect(dnsLinkResolver.callCount).to.equal(0) // not called because parseResource never passes the resource to parseUrlString
-    expect(blockRetriever.retrieve.callCount).to.equal(0) // not called because the blockstore has the content
-    expect(unixFsStatStub.callCount).to.equal(1)
-    expect(unixFsCatStub.callCount).to.equal(0) // not called because the request was aborted during .stat call
+    // file root
+    await expect(helia.blockstore.has(fileCid))
+      .to.eventually.be.true()
+
+    // leaf nodes
+    await expect(helia.blockstore.has(leaf1))
+      .to.eventually.be.true()
+    await expect(helia.blockstore.has(leaf2))
+      .to.eventually.be.true()
+
+    const fileRootGot = pDefer()
+    const blockstoreGetSpy = Sinon.stub(helia.blockstore, 'get')
+    blockstoreGetSpy.callsFake(async (cid, options) => {
+      if (cid.equals(fileCid)) {
+        fileRootGot.resolve()
+      }
+
+      return blockstoreGetSpy.wrappedMethod.call(helia.blockstore, cid, options)
+    })
```

Comment:
I was not familiar with `wrappedMethod`, nice.

Reply:
Me neither, tbh.  It doesn't bind the right context so we have to use `.call` or `.apply` but it seems more concise than creating closures or whatever.